pkg/krmtotf/fetchlivestate.go

```diff
@@ -53,15 +53,12 @@ func FetchLiveState(ctx context.Context, resource *Resource, provider *tfschema.
 
 }
 
-// Special handling for KMSCryptoKey that still lives after its parent KMSKeyRing is deleted.
+// SkipOrphanedCheck Special handling for KMSCryptoKey that still lives after its parent KMSKeyRing is deleted.
 // We can import the tf state directly from itself instead of sourcing for its parent.
 // More info in b/279485255#comment14
-func shouldGetImportIDFromSelfForDelete(resource *Resource) bool {
-	return resource.Kind == "KMSCryptoKey"
-}
-
-func ShouldResolveParentForDelete(resource *Resource) bool {
-	return !shouldGetImportIDFromSelfForDelete(resource) || hasEmptySelfLink(resource)
+func SkipOrphanedCheck(resource *Resource) bool {
```

Comment:
Similar to [this comment](https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1068/files#r1454250262), I would prefer to negate the condition when we handle it instead of in the helper function.